frontend/isoko/src/pages/SearchResults.tsx

```diff
@@ -0,0 +1,402 @@
+   const removeTag = (business, tag) => {
+      const tags = business.tags;
+      const selectedTags = tagState.filter((t) => t.selected == true);
+
+      const index = selectedTags.map((t) => t.text).indexOf(tag);
+      selectedTags.splice(index, 1);
+
+      // if the tag to remove isn't associated with this business, it should still be displayed
+      if (tags.indexOf(tag) === -1) {
+         return true;
+      }
+
+      // if the tag to remove is the only one associated with this business, it should be removed from results
+      if (tags.length === 1) {
+         return false;
+      }
+
+      // if there are more minorities to filter by, check if other tags associated with business fulfill them
+      if (selectedTags.length > 0) {
+         return selectedTags.some((t) => tags.includes(t.text));
+      }
+   };
+
+   // filter businesses based on time
+   const filterBusinessesByTime = (e, businesses) => {
+      const day = dayOfWeek();
+      // filter by time if box is checked, if unchecked, revert to original results
+      if (e.target.checked) {
+         setFilteredBusinesses(businesses.filter((b) => filterByTime(b, day)));
+      } else {
+         setFilteredBusinesses(businessResults);
+      }
+   };
+
+   // filter business results when customer removes a tag
+   const filterBusinessesRemoveTag = (key, businesses) => {
+      setFilteredBusinesses(businesses.filter((b) => removeTag(b, key)));
+      const updatedTagState = [...tagState];
+      const idx = tagState.map((t) => t.text).indexOf(key);
+      updatedTagState[idx].selected = !updatedTagState[idx].selected;
+      setTagState(updatedTagState);
+   };
+
+   // sort businesses by value of sort dropdown
+   const sortBusinesses = (key) => {
+      if (key === 'reviews') {
+         setFilteredBusinesses([
+            ...filteredBusinesses.sort(
+               (a, b) => b.reviews.length - a.reviews.length
+            ),
+         ]);
+      } else if (key === 'recent') {
+         setFilteredBusinesses([
+            ...filteredBusinesses.sort((a, b) => a.timestamp - b.timestamp),
+         ]);
+      }
+   };
+
+   // this function will requery with new tags
+   // already added to props.minorityTags
+   const applyNewTags = (e, newTags) => {
+      // toggle tags if previously selected and were deleted from modal, or if not selected and added through the modal
+      tagState.forEach((t, i) => {
+         if (
+            (t.selected && !newTags.includes(t.text)) ||
+            (!t.selected && newTags.includes(t.text))
+         ) {
+            const updatedTagState = [...tagState];
+            updatedTagState[i].selected = !updatedTagState[i].selected;
+            setTagState(updatedTagState);
+         }
+      });
+
+      setShowModal(false);
+   };
+
+   return (
+      <SearchContainer>
+         <Row>
+            <Col className="col-sm-3">
```

Comment:
nit: for consistency can we change this to this
```suggestion
            <Col sm={3}>
```

Reply:
fixing these made it ignore your approval smh had to re-request